### ISSUE TITLE
[bug] fundrawtransaction will change output address and value position.

### ISSUE THREAD

**Issue by tinshen** [dashpay/dash#5666]
Vernon 19.2.0
step1: createrawtransaction 2 output
{a1:a2,b1:b2}
step2:fundrawtransaction changeAddress c1

step3
{a1:b2,b1:a2,c1:c2}

createrawtransaction_tx: 0200000001adee85fed236e0ffd5bbcaa92309ea8c4c72caabbe87cb7d8aa2c65b1c3ac1260200000000ffffffff0260b03f05000000001976a91403f17b0f5c26129b929a6e44281b855c973ad7e688acd0a11000000000001976a914609000121bf257990ebcfaefa9054e1a5288ce1988ac00000000

fundrawtransaction_tx:0200000001adee85fed236e0ffd5bbcaa92309ea8c4c72caabbe87cb7d8aa2c65b1c3ac1260200000000ffffffff03d0a11000000000001976a91403f17b0f5c26129b929a6e44281b855c973ad7e688ac60b03f05000000001976a914609000121bf257990ebcfaefa9054e1a5288ce1988acf0c7c8ad010000001976a914bae57e05a75e4eca5e7150a56682796357a5bb1e88ac00000000

**Comment by knst** [comment]
@tinshen thanks for report, it seems as it indeed can happen sometime:

```
{'txid': '9bb2f44340e395f2063ca65b538e758fb520888433b5bb2f05ca0975cece18b6', 'version': 2, 'type': 0, 'size': 78, 'locktime': 0, 'vin': [], 'vout': [{'value': Decimal('1.00000000'), 'valueSat': 100000000, 'n': 0, 'scriptPubKey': {'asm': 'OP_DUP OP_HASH160 411cced0d6b0ac6f6e7c974e7007024fe93f5266 OP_EQUALVERIFY OP_CHECKSIG', 'hex': '76a914411cced0d6b0ac6f6e7c974e7007024fe93f526688ac', 'reqSigs': 1, 'type': 'pubkeyhash', 'addresses': ['ySFjHH6BQQ9WEo52MakwrZHoaTbq9RLJo2']}}, {'value': Decimal('0.10000000'), 'valueSat': 10000000, 'n': 1, 'scriptPubKey': {'asm': 'OP_DUP OP_HASH160 8132c231e7e9758399e1e3c1ac2be38e4e72c6b8 OP_EQUALVERIFY OP_CHECKSIG', 'hex': '76a9148132c231e7e9758399e1e3c1ac2be38e4e72c6b888ac', 'reqSigs': 1, 'type': 'pubkeyhash', 'addresses': ['yY6apiXLRPwCbBDWBM6DCRHKrULfUvPDZB']}}]}
```
after fund transaction:
```
{'txid': '9916d0914b25376c1c5f32e903845268b6fd64f1d2f2c15f624d4b3cc713d5de', 'version': 2, 'type': 0, 'size': 153, 'locktime': 0, 'vin': [{'txid': '6d2474ee4b88feff91faa1666451d11c3de0c7c39d2b50be88279db5094f3948', 'vout': 1, 'scriptSig': {'asm': '', 'hex': ''}, 'sequence': 4294967294}], 'vout': [{'value': Decimal('0.10000000'), 'valueSat': 10000000, 'n': 0, 'scriptPubKey': {'asm': 'OP_DUP OP_HASH160 411cced0d6b0ac6f6e7c974e7007024fe93f5266 OP_EQUALVERIFY OP_CHECKSIG', 'hex': '76a914411cced0d6b0ac6f6e7c974e7007024fe93f526688ac', 'reqSigs': 1, 'type': 'pubkeyhash', 'addresses': ['ySFjHH6BQQ9WEo52MakwrZHoaTbq9RLJo2']}}, {'value': Decimal('1.00000000'), 'valueSat': 100000000, 'n': 1, 'scriptPubKey': {'asm': 'OP_DUP OP_HASH160 8132c231e7e9758399e1e3c1ac2be38e4e72c6b8 OP_EQUALVERIFY OP_CHECKSIG', 'hex': '76a9148132c231e7e9758399e1e3c1ac2be38e4e72c6b888ac', 'reqSigs': 1, 'type': 'pubkeyhash', 'addresses': ['yY6apiXLRPwCbBDWBM6DCRHKrULfUvPDZB']}}, {'value': Decimal('430.02044190'), 'valueSat': 43002044190, 'n': 2, 'scriptPubKey': {'asm': 'OP_DUP OP_HASH160 dbca309cc225c6ea313df574af91724f3a321f3c OP_EQUALVERIFY OP_CHECKSIG', 'hex': '76a914dbca309cc225c6ea313df574af91724f3a321f3c88ac', 'reqSigs': 1, 'type': 'pubkeyhash', 'addresses': ['ygMb3cZQ3PNKWHAUMkjNKs1rVmvx8yNsfG']}}]}
```

I am checking implementation

**Comment by UdjinM6** [comment]
@tinshen Thanks for reporting! 👍  pls test #5667

**Comment by PastaPastaPasta** [comment]
Resolved in merged #5667
